arangod/StorageEngine/VPackSortMigration.cpp

```diff
@@ -0,0 +1,294 @@
+                  VPackValue("all good with sorting order"));
+    } else {
+      guardO->add(StaticStrings::Error, VPackValue(true));
+      guardO->add(StaticStrings::ErrorCode,
+                  VPackValue(TRI_ERROR_ARANGO_INDEX_HAS_LEGACY_SORTED_KEYS));
+      guardO->add(StaticStrings::ErrorMessage,
+                  VPackValue("some indexes have legacy sorted keys"));
+    }
+  }
+
+  return {};
+}
+
+Result migrateVPackIndexSorting(TRI_vocbase_t& vocbase, VPackBuilder& result) {
+  // Just for the sake of completeness, restrict ourselves to the RocksDB
+  // storage engine:
+  result.clear();
+  auto& engineSelectorFeature =
+      vocbase.server().getFeature<EngineSelectorFeature>();
+  if (!engineSelectorFeature.isRocksDB()) {
+    {
+      VPackObjectBuilder guard(&result);
+      guard->add(StaticStrings::Error, VPackValue(false));
+      guard->add(StaticStrings::ErrorCode, VPackValue(0));
+      guard->add(
+          StaticStrings::ErrorMessage,
+          VPackValue("VPack sorting migration is unnecessary for storage "
+                     "engines other than RocksDB"));
+    }
+    return {};
+  }
+
+  auto& engine = engineSelectorFeature.engine<RocksDBEngine>();
+  Result res = engine.writeSortingFile(
+      arangodb::basics::VelocyPackHelper::SortingMethod::Correct);
+
+  if (res.ok()) {
+    VPackObjectBuilder guard(&result);
+    guard->add(StaticStrings::Error, VPackValue(false));
+    guard->add(StaticStrings::ErrorCode, VPackValue(0));
+    guard->add(StaticStrings::ErrorMessage,
+               VPackValue("VPack sorting migration done."));
+    return {};
+  }
+  return res;
+}
+
+namespace {
+async<Result> fanOutRequests(TRI_vocbase_t& vocbase, fuerte::RestVerb verb,
+                             VPackBuilder& result, bool includeAgents) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+
+  auto& ci = vocbase.server().getFeature<ClusterFeature>().clusterInfo();
+  auto& nf = vocbase.server().getFeature<NetworkFeature>();
+
+  auto dbs = ci.getCurrentDBServers();
+  std::vector<futures::Future<network::Response>> requests;
+  requests.reserve(dbs.size());
+
+  network::RequestOptions opts;
+  opts.database = vocbase.name();
+
+  std::string path =
+      absl::StrCat("/_admin/cluster/vpackSortMigration/",
+                   verb == fuerte::RestVerb::Get ? "check" : "migrate");
+  for (auto const& server : dbs) {
+    auto f = network::sendRequest(nf.pool(), "server:" + server, verb, path, {},
+                                  opts);
+    requests.emplace_back(std::move(f).then([server](auto&& response) {
+      LOG_DEVEL << "received response from " << server;
+      return std::move(response).get();
+    }));
+  }
+
+  LOG_TOPIC("22535", DEBUG, Logger::ENGINES)
+      << "VPackSortMigration: awaiting all results";
+  auto responses = co_await futures::collectAll(requests);
+  LOG_TOPIC("22536", DEBUG, Logger::ENGINES)
+      << "VPackSortMigration: all server responded";
+
+  auto server = dbs.begin();
```

Comment:
This looks a bit problematic, because it assumes that `dbs` contains the same amount of values as `responses`.
I am not sure if this is always guaranteed.
I am also not sure if the order of responses is guaranteed to be the same as the order of requests we sent.

Instead of using `dbs` here, we could simple use the `destination` member from each response object, which also contains the target server id.